examples/tools/helloworld.ts

```diff
@@ -0,0 +1,20 @@
+import { BaseToolOptions, BaseToolRunOptions, StringToolOutput, Tool } from "@/tools/base.js";
+import { z } from "zod";
+
+type ToolOptions = BaseToolOptions;
+type ToolRunOptions = BaseToolRunOptions;
+
+export class HelloWorldTool extends Tool<StringToolOutput, ToolOptions, ToolRunOptions> {
+  name = "Helloworld";
+  description = "Says hello when asked for a special greeting.";
+
+  inputSchema = z.string();
+
+  static {
+    this.register();
+  }
+
+  protected async _run(input: string, options?: BaseToolRunOptions): Promise<StringToolOutput> {
```

Comment:
`protected async _run(input: string, options?: ToolRunOptions): Promise<StringToolOutput> {`

Reply:
or don't provide any generics, but first

Reply:
Resolved (pending further review) in f6237ef